EgammaAnalysis/ElectronTools/interface/EGEnergySysIndex.h

```diff
@@ -0,0 +1,32 @@
+class EGEnergySysIndex {
```

Comment:
Please use an include guard matching the full file name subsystem_package_fileName_h

Reply:
So opps about the include guard, my bad here. 

The problem with your two changes is that I want to use it to solely access vectors and enum class would cause me to have to cast to an integer each time. Likewise for the size_t, I'm using this solely to compare to std::vector::size() hence  the size_t. I mean I could swap it to an int but when dealing with vector::size() I tend to use size_t as a rule.

Reply:
ok for now about scoping/size_t.
hope not to come back to this later.